monix-execution/jvm/src/main/scala/monix/execution/schedulers/TracingScheduler.scala

```diff
@@ -0,0 +1,89 @@
+package monix.execution.schedulers
+
+import java.util.concurrent.{ScheduledExecutorService, TimeUnit}
+import monix.execution.misc.Local
+import monix.execution.{Cancelable, Scheduler, UncaughtExceptionReporter, ExecutionModel => ExecModel}
+import scala.concurrent.ExecutionContext
+
+final class TracingScheduler private (
+                                       scheduler: ScheduledExecutorService,
```

Comment:
I think it would be a better idea to wrap our own `SchedulerService`, otherwise you're restricting instances to just wrap `ScheduledExecutorService` and for example that excludes `ForkJoinPool` (AFAIK, not absolutely sure).